### ISSUE TITLE
Compact nested mapping/sequence requires extra leading spaces

### ISSUE THREAD

**Issue by roller** [jeremyfa/yaml.js#9]
I'm not sure if this is a matter of YAML spec, or just a difference between other parsers, but yamljs appears to require extra leading spaces for nested mappings.

``` yaml

---
test:
- a:
  - b:
    - one
```

yaml2json returns the following:

```
Unable to parse in "it.yaml" at line 5 (near "    - one").
```

Compared to [pyyaml](http://yaml-online-parser.appspot.com/?yaml=---%0Atest%3A%0A-+a%3A%0A++-+b%3A%0A++++-+one%0A%0A&type=canonical_yaml) which parses ok.

The required spacing appears to depend on the level of nesting.  For instance this works (further indented -a)

``` yaml

---
test:
 - a:
   - b:
     - one
```

But this returns "Unable to parse".

``` yaml

---
test:
 - a:
   - b:
     - c:
       - one
```

**Comment by jeremyfa** [comment]
Yes, this is a known issue. Not so simple to fix with the current implementation but I plan to revise the code soon to handle these cases.

**Comment by jeremyfa** [comment]
This should be fixed in `v0.2.0`
